src/coin_test/analysis/graphs.py

```diff
@@ -0,0 +1,516 @@
+"""Generate plots for analysis."""
+
+from abc import ABC, abstractmethod
+import math
+from typing import Sequence
+
+import datapane as dp
+import pandas as pd
+from plotly.colors import n_colors
+import plotly.graph_objects as go
+
+from coin_test.backtest import TradeRequest
+from coin_test.backtest.backtest_results import BacktestResults
+from coin_test.util import Side
+
+
+def _get_lims(figs: Sequence[go.Figure], x: bool = True) -> list[tuple[float, float]]:
+    x_vals, y_vals = [], []
+    for fig in figs:
+        full_fig = fig.full_figure_for_development(warn=False)
+        x_vals.extend(full_fig.layout.xaxis.range)
+        y_vals.extend(full_fig.layout.yaxis.range)
+    ret = []
+    if x:
+        ret.append((min(x_vals), max(x_vals)))
+    ret.append((min(y_vals), max(y_vals)))
+    return ret
+
+
+def _flatten_strategies(results: BacktestResults) -> str:
+    return "-".join(results.strategy_names)
+
+
+def _get_strategies(results: Sequence[BacktestResults]) -> list[str]:
+    return sorted(list(set(_flatten_strategies(r) for r in results)))
```

Comment:
Sorting is arbitrary, although does make the rest of the code more consistent w.r.t. how strategies are ordered. The set call is required since there will be multiple results with the same strategy.